node/src/rpc.rs

```diff
@@ -47,8 +55,13 @@ impl NodeRpcServer for NodeRpc {
     fn version(&self) -> rpc::Result<String> {
         Ok(env!("CARGO_PKG_VERSION").into())
     }
+
+    fn set_mock_time(&self, time: u64) -> rpc::Result<()> {
+        crate::mock_time::set_mock_time(self.chain_type, time)?;
+        Ok(())
+    }
 }
 
-pub fn init(shutdown_trigger: ShutdownTrigger) -> rpc::Methods {
-    NodeRpc::new(shutdown_trigger).into_rpc().into()
+pub fn init(shutdown_trigger: ShutdownTrigger, chain_type: ChainType) -> rpc::Methods {
```

Comment:
Perhaps it's better to pass an `Arc<ChainConfig>` here... no need to be so conservative about it.

Reply:
Sure, updated.